crates/interpreter/src/valid.rs

```diff
@@ -793,45 +805,52 @@ impl<'a, 'm> Expr<'a, 'm> {
     }
 
     fn stitch(&mut self, source: SideTableBranch, target: SideTableBranch) {
-        let SideTableBranch { parser, side_table } = source;
+        let SideTableBranch { parser, side_table, val_cnt, pop_cnt } = source;
         let entry = &mut self.side_table[side_table];
         assert!(entry.is_none());
         *entry = Some(SideTableEntryView {
             delta_ip: target.delta_ip(parser),
             delta_stp: target.delta_stp(side_table),
-            // TODO(dev/fast-interp): Compute the fields below.
-            val_cnt: 0,
-            pop_cnt: 0,
+            val_cnt,
+            pop_cnt,
```

Comment:
I think this should just be `(source.stack - target.stack) as u32` or something like that. We should actually check that the subtraction and the cast are not overflowing, and return `Unsupported` otherwise.

We should probably also remove `val_cnt` from `source.stack` (again checking for overflow).